mlrun/feature_store/feature_set.py

```diff
@@ -488,6 +489,31 @@ def set_targets(
         if default_final_step:
             self.spec.graph.final_step = default_final_step
 
+    def validate_steps(self):
+        if not self.spec:
+            return
+        if not self.spec.graph:
+            return
+        for step in self.spec.graph.steps.values():
+            if (
+                step.class_name in queue_class_names
+                or step.class_name is None
+                or "." not in step.class_name
+            ):
+                #  we are not checking none class names or queue class names.
+                continue
+            module_path, class_name = step.class_name.rsplit(".", 1)
+            if not module_path or not class_name:
+                continue
+            module = importlib.import_module(module_path)
+            step_class = getattr(module, class_name)
+            if not hasattr(step_class, "validate"):
+                continue
+            if not step.class_args:
+                continue
```

Comment:
Why continue? It should be valid to have empty `class_args`, just make sure you handle it correctly when creating the `step_object`